data/Makefile.am

```diff
@@ -7,17 +7,20 @@ xml_files = $(xml_in_files:.xml.in=.xml)
 	$(AM_V_GEN) GETTEXTDATADIR=$(top_srcdir) $(MSGFMT) --xml --template $< -d $(top_srcdir)/po -o $@
 
 desktopdir = $(datadir)/applications
-desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
-desktop_in_files = \
-	caja.desktop.in \
-	caja-home.desktop.in \
-	caja-computer.desktop.in \
-	caja-folder-handler.desktop.in \
-	caja-file-management-properties.desktop.in  \
-	caja-browser.desktop.in                     \
-	caja-autorun-software.desktop.in
-%.desktop: %.desktop.in
+desktop_DATA = \
+	caja.desktop \
+	caja-home.desktop \
+	caja-computer.desktop \
+	caja-folder-handler.desktop \
+	caja-file-management-properties.desktop  \
+	caja-browser.desktop \
+	caja-autorun-software.desktop
+desktop_in_files = $(desktop_DATA:.desktop=.desktop.in)
+desktop_in_in_files = $(desktop_DATA:.desktop=.desktop.in.in)
+%.desktop.in: %.desktop.in.in
 	$(AM_V_GEN) GETTEXTDATADIR=$(top_srcdir) $(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
+%.desktop: %.desktop.in
+	$(AM_V_GEN) sed -e 's|@bindir[@]|$(bindir)|g' $< > $@
```

Comment:
`$(bindir)` is an [automake variable](https://www.gnu.org/prep/standards/html_node/Directory-Variables.html), you can set its value, for instance:
```shell
$ make clean
$ mkdir -p /home/robert/bin
$ ./autogen.sh --bindir=/home/robert/bin && make
$ cat data/*.desktop | grep Exec=
TryExec=/home/robert/bin/caja-autorun-software
Exec=/home/robert/bin/caja-autorun-software %u
TryExec=/home/robert/bin/caja
Exec=/home/robert/bin/caja --no-desktop --browser %U
TryExec=/home/robert/bin/caja
Exec=/home/robert/bin/caja --no-desktop computer:
Exec=/home/robert/bin/caja
Exec=/home/robert/bin/caja-file-management-properties
TryExec=/home/robert/bin/caja
Exec=/home/robert/bin/caja --no-desktop %U
TryExec=/home/robert/bin/caja
Exec=/home/robert/bin/caja --no-desktop
```
Otherwise, you can always override the value using sed, like in Makefile.am

Reply:
This change is based on http://git.savannah.gnu.org/cgit/gettext.git/tree/gettext-tools/examples/hello-c-gnome3/

Reply:
@raveit65 - Do you remember if it's NixOS?

Reply:
yes, it was a PR for NixOS